website/src/controllers/groups.js

```diff
@@ -529,6 +529,15 @@ api.leave = function(req, res, next) {
         update['$unset']['quest.members.' + user._id] = 1;
       }
       // FIXME do we want to remove the group `if group.members.length == 0` ? (well, 1 since the update hasn't gone through yet)
+	  //The following addition deletes party or private guild in case the user leaving the party is the last one
+	  if(group.type === 'party' && group.members.length == 1 || group.type === 'guild' && group.privacy === 'private' && group.members.length == 1 ) {
```

Comment:
I would wrap the statements before and after the OR (||) in parenthesis, I'm not sure this way they work correctly and anyway it will help with readability